aiida/orm/implementation/general/calculation/job/__init__.py

```diff
@@ -118,6 +118,28 @@ def process(cls):
     def get_builder(cls):
         return cls.process().get_builder()
 
+    def get_builder_restart(self):
+        from aiida.work.job_processes import JobProcess
+        from aiida.work.ports import PortNamespace
+
+        inputs = self.get_inputs_dict()
+        options = self.get_options()
+        builder = self.get_builder()
+
+        for port_name, port in self.process().spec().inputs.items():
+            if port_name == JobProcess.OPTIONS_INPUT_LABEL:
```

Comment:
Does this work? This does `builder.options = {...}`, can this be done directly or should we loop?